### PR TITLE
Support gn-format hook in all-repos.yaml

### PR DESCRIPTION
**Pull request by KindDragon**
my new repository:

- [x] is added to the bottom *or* with existing repos from the same account
- [x] contains a license
- [x] is not `language: system`, `language: script`, `language: docker`, or `language: docker_image`
- [x] does not contain "pre-commit" in the name

This hook for formatting GN build files https://gn.googlesource.com/gn/

### CODE REVIEW

File: all-repos.yaml [hunk 1]
```diff
@@ -244,3 +244,4 @@
 - https://github.com/numpy/numpydoc
 - https://github.com/hakancelikdev/unimport
 - https://github.com/hakancelikdev/unexport
+- https://github.com/KindDragon/gn-build-py
```
Comment (asottile) [inline all-repos.yaml:247]: just a few things:
- `types: [file]` is the default so you can leave that out: https://github.com/KindDragon/gn-build-py/blob/d85d1eded700c76a912a7ca38f8d4963dbb995e4/.pre-commit-hooks.yaml#L6C3-L6C16
- `rev: latest` (1) does not work and (2) [is not supported](https://pre-commit.com/#using-the-latest-version-for-a-repository): https://github.com/KindDragon/gn-build-py/blame/d85d1eded700c76a912a7ca38f8d4963dbb995e4/README.md#L11

Comment (KindDragon) [reply all-repos.yaml:247]: I've made corrections based on your suggestions

Comment (asottile) [reply all-repos.yaml:247]: `rev: ''` is also similarly not supported

Comment (KindDragon) [reply all-repos.yaml:247]: Yes, but I added to README.md to call `pre-commit autoupdate` after that to use latest available revision

**Pull request merged**